ibc/utils.go

```diff
@@ -43,3 +43,22 @@ func GetTransferSenderRecipient(packet channeltypes.Packet) (
 
 	return sender, recipient, data.Sender, data.Receiver, nil
 }
+
+// GetTransferAmount returns the amount from an ICS20 FungibleTokenPacketData.
+func GetTransferAmount(packet channeltypes.Packet) (
+	amount sdk.Int,
+	err error,
+) {
+	// unmarshal packet data to obtain the sender and recipient
+	var data transfertypes.FungibleTokenPacketData
+	if err := transfertypes.ModuleCdc.UnmarshalJSON(packet.GetData(), &data); err != nil {
+		return sdk.ZeroInt(), sdkerrors.Wrapf(sdkerrors.ErrUnknownRequest, "cannot unmarshal ICS-20 transfer packet data")
+	}
+
+	amount, ok := sdk.NewIntFromString(data.Amount)
```

Comment:
hmm not sure how it's handled in IBC-go when chains have a different precision, we need to double-check it (6 vs 18)